### PR TITLE
runtime: add --all option to kill command

### LINKED ISSUE #693: Add --all option for kill command
`runc` added a `--all`l option to the `kill` command on commit https://github.com/opencontainers/runc/commit/e58671e530eb04a233d953c6a56e56804703f80c. This is being used by docker 1.13 so we also need to support `--all`.

### PR DESCRIPTION
**Pull request by devimc**
With this patch users will can send signal to
all processes running inside the container

fixes #693

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Comment by chavafg** [comment]
qa-failed

**Comment by chavafg** [comment]
qa-passed

**Comment by devimc** [comment]
waiting for https://github.com/clearcontainers/hyperstart/pull/9

**Comment by chavafg** [comment]
qa-passed

File: src/commands/kill.c [hunk 1]
```diff
@@ -23,6 +23,19 @@
 
 #include "command.h"
 
+static gboolean allProcesses = false;
+
+static GOptionEntry options_kill[] =
+{
+	{
+		"all", 'a' , G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_NONE, &allProcesses,
```
Comment (dlespiau) [inline src/commands/kill.c]: all_processes. CamelCase is usually only for types in the glib/GNOME world, variables remain with a _

**Comment by dlespiau** [comment]
With the naming convention remark addressed:

lgtm

File: src/commands/kill.c (hunk 1 above)
Comment (devimc) [inline src/commands/kill.c]: ouch, thanks

**Comment by chavafg** [comment]
qa-passed

**Comment by dlespiau** [comment]
@jodh-intel Can I ask for a favour and your help reviewing this PR? Julio is gatekeeper this week and can't lgtm his own work.

File: tests/functional/kill.bats [hunk 2]
```diff
@@ -144,3 +121,35 @@ function teardown() {
 	run_cmd "$cmd" "0" "$COR_TIMEOUT"
 	verify_runtime_dirs "$container_id" "deleted"
 }
+
+@test "start then kill --all" {
+	signal=SIGTERM
+	trap_file=/trap.log
+	workload_cmd \
+		"sh" "-c" \
+		"trap 'touch ${trap_file} ; exit 0' $signal \&\& while : ; do sleep 1; done"
```
Comment (jodh-intel) [inline tests/functional/kill.bats]: The joys of `\q\u\o\t\i\n\g`! :smile:

**Comment by jodh-intel** [comment]
lgtm

File: tests/functional/kill.bats (hunk 2 above)
Comment (devimc) [inline tests/functional/kill.bats]: I can't quote it ;(  docker replaces ```&``` with ```\u0026```

**Comment by dlespiau** [comment]
1/0 approved says pullapprove being its usual Sibylline self.

```
qa		1 / 0	Approved by chavafg
```

@chavafg mind approving that one again?

**Comment by chavafg** [comment]
qa-passed

**Comment by chavafg** [comment]
I think it is already approved

**Comment by dlespiau** [comment]
pullapprove fails again (the merge button is grayed out)

File: tests/functional/kill.bats (hunk 2 above)
Comment (jodh-intel) [inline tests/functional/kill.bats]: Yeah, I know. I wasn't suggesting you change it - seeing the backslashed aphersands just made me laugh. Maybe I need to get out more ;-)

**Comment by devimc** [comment]
lgtm

**Comment by GabyCT** [comment]
qa-passed

**Comment by dlespiau** [comment]
Alright, this thing is busted :) had to disable the branch protection temporarily to override it. All good, thanks all.

**Comment by devimc** [comment]
thanks @dlespiau

**Pull request merged**